.github/workflows/ci.yml

```diff
@@ -0,0 +1,22 @@
+name: CI
```

Comment:
Please, use Play [sharable workflow](https://github.com/playframework/.github) for CI 🙏 
See https://github.com/playframework/play-ws/blob/main/.github/workflows/build-test.yml as example 😉

Reply:
@ihostage Is there a way to use sharable workflow with macos image?  I think macos image can run best with our tests here with support of the most browser types.

Reply:
Hmmm 🤔 What will we lose if we use Linux image?

Reply:
@ihostage We have selenium tests here that test for different types of browser, using the macos image we can run the tests for safari, chrome, firefox and msedge in addition to htmlunit, only the old internet explorer tests will be skipped.  Using linux I think we can only run chrome, firefox and msedge.

Reply:
@ihostage Perhaps I can keep the test job but try to use the sharable workflow for check-code-style, check-binary-compatibility, check-docs?

Reply:
@ihostage I studied the sharable workflow source a bit, the matrix approach looks nice, I wonder if it is possible to make the runs-on as an input with default value of 'ubuntu-20.04', but allow here in scalatest+play to override it as 'macos-latest' ?

Reply:
@ihostage Trying out the check-docs, but encountered this weird error: 

![Screenshot from 2022-09-21 12-00-40](https://user-images.githubusercontent.com/1322009/191411509-7be7f01c-4195-4bc4-91a1-a6235f64f975.png)

Mind to help seeing what's going wrong?  fyi I am trying out the sharable workflow in this PR: 

https://github.com/playframework/scalatestplus-play/pull/373